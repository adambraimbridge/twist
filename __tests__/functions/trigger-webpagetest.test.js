/* global expect,jest */
/* eslint-disable no-console */

// This block must go first
global.console = {log: jest.fn()};
const lambdaTester = require('lambda-tester');
const handler = require('../../functions/trigger-webpagetest').handler;
const fetch = require('node-fetch');

// You can edit after here.
const payload = require('../../__mocks__/fixtures/trigger-webpagetest.json');

describe('functions/trigger-webpagetest', () => {
	describe('given a URL', () => {
		test('it should trigger a Web Page Test for that URL', () => {
			return lambdaTester(handler)
				.event(payload)
				.expectResult(response => {
					expect(fetch).toHaveBeenCalled();
					expect(response).toMatchSnapshot();
				});
		});
	});
});
