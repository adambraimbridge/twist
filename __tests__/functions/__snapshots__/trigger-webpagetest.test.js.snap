// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions/trigger-webpagetest given a URL it should trigger a Web Page Test for that URL 1`] = `
Object {
  "body": "Twist: Sucessfully triggered Webpagetools test: https://www.webpagetest.org/result/171116_QW_768975393ffe36d0fd0f76ac9f1ab902/",
  "statusCode": 200,
}
`;
